sympy/solvers/ode/single.py

```diff
@@ -791,66 +791,93 @@ def _get_general_solution(self, *, simplify: bool = True):
-    >>> x = symbols('x')
+    >>> x, eq = symbols('x eq')
+
+    >>> eq = f(x).diff(x, x) + f(x).diff(x)**2
+    >>> dsolve(eq, hint='2nd_autonomous_nth')
+    Eq(f(x), log(C1*(C2 + x)))
 
-    >>> eq = f(x).diff(x, 2) - g(f(x))
-    >>> dsolve(eq, simplify=False)
-    [Eq(Integral(1/sqrt(C1 + 2*Integral(g(_u), _u)), (_u, f(x))), C2 + x),
-    Eq(Integral(1/sqrt(C1 + 2*Integral(g(_u), _u)), (_u, f(x))), C2 - x)]
+    >>> eq = f(x).diff(x, x) - f(x).diff(x)**2
+    >>> dsolve(eq, hint='2nd_autonomous_nth')
+    Eq(f(x), log(C1/(C2 + x)))
 
-    >>> from sympy import exp, log
+    >>> from sympy import exp, log, cbrt
     >>> eq = f(x).diff(x, 2) - exp(f(x)) + log(f(x))
-    >>> dsolve(eq, simplify=False)
-    [Eq(Integral(1/sqrt(-2*_u*log(_u) + 2*_u + C1 + 2*exp(_u)), (_u, f(x))), C2 + x),
-    Eq(Integral(1/sqrt(-2*_u*log(_u) + 2*_u + C1 + 2*exp(_u)), (_u, f(x))), C2 - x)]
+    >>> dsolve(eq, hint='2nd_autonomous_nth', simplify=False)
+    [Eq(Integral(1/sqrt(-2*_v*log(_v) + 2*_v + C1 + 2*exp(_v)), (_v, f(x))), C2 + x),
+    Eq(Integral(1/sqrt(-2*_v*log(_v) + 2*_v + C1 + 2*exp(_v)), (_v, f(x))), C2 - x)]
+
+    >>> eq = f(x).diff(x, 2) + cbrt(f(x)) + 1/f(x)
+    >>> dsolve(eq, hint='2nd_autonomous_nth', simplify=False)
+    [Eq(sqrt(2)*Integral(1/sqrt(-3*_v**(4/3) + 2*C1 - 4*log(_v)), (_v, f(x))), C2 + x),
+    Eq(sqrt(2)*Integral(1/sqrt(-3*_v**(4/3) + 2*C1 - 4*log(_v)), (_v, f(x))), C2 - x)]
+
+    >>> eq = f(x).diff(x, x) + f(x).diff(x)**4
+    >>> dsolve(eq, hint='2nd_autonomous_nth', simplify=False)
+    [Eq((C1 + 2*f(x))**(3/2)/3, C2 + x), Eq((C1 + 2*f(x))**(3/2)/3, C2 - x)]
 
     References
     ==========
 
-    http://eqworld.ipmnet.ru/en/solutions/ode/ode0301.pdf
+    https://en.wikipedia.org/wiki/Autonomous_system_(mathematics)
     """
-    hint = "2nd_nonlinear_autonomous_conserved"
+
+    hint = "2nd_autonomous_nth"
     has_integral = True
     order = [2]
 
     def _wilds(self, f, x, order):
-        fy = Wild('fy', exclude=[0, f(x).diff(x), f(x).diff(x, 2)])
-        return (fy,)
+        a = Wild('a', exclude=[0, f(x).diff(x), f(x).diff(x, 2)])
+        n = Wild('n', exclude=[x, f(x), f(x).diff(x), f(x).diff(x, 2)])
+        return a, n
 
     def _equation(self, fx, x, order):
-        fy = self.wilds()[0]
-        return fx.diff(x, 2) + fy
-
-    def _verify(self, fx):
-        return self.ode_problem.is_autonomous
+        a, n = self.wilds()
+        return fx.diff(x, 2) + a*fx.diff(x)**n
 
     def _get_general_solution(self, *, simplify: bool = True):
-        g = self.wilds_match()[0]
+        a, n = self.wilds_match()
         fx = self.ode_problem.func
         x = self.ode_problem.sym
-        u = Dummy('u')
-        g = g.subs(fx, u)
-        C1, C2 = self.ode_problem.get_numbered_constants(num=2)
-        inside = -2*Integral(g, u) + C1
-        lhs = Integral(1/sqrt(inside), (u, fx))
-        return [Eq(lhs, C2 + x), Eq(lhs, C2 - x)]
-
+        (C1, C2, C3) = self.ode_problem.get_numbered_constants(num=3)
```

Comment:
Why does this need 3 constants? There should be only 2 constants for a 2nd order ODE.

Reply:
My bad, I will correct it.